app/views/stories/show.html.erb

```diff
@@ -22,7 +22,11 @@
         <p><%= render partial: 'ratings/rate', locals: { story: @story } %></p>
       <% end %>
       <% if @story.tier == 'paid' %>
-        <%= render 'charges/charge', story_id: @story.id %>
+        <% if current_user && current_user.purchase?(@story) %>
```

Comment:
I need to re-organize these permissions, left it as is for now to show it all working. Ideally a user won't be able to add paid stories to their collection or rate them until they purchase it. I'll do a re-draw of the show page eventually.